com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/util/ApplicationURITest.java

```diff
@@ -24,9 +34,100 @@
+    }
+
+    private static Stream<Arguments> routeTestParameters() {
+        return Stream.of(Arguments.of(createCloudRoute("custom-host", createCloudDomain("custom-domain"), null), "custom-host",
+                                      "custom-domain", null),
+                         Arguments.of(createCloudRoute("custom-host", createCloudDomain("custom-domain"), "/custom-path"), "custom-host",
+                                      "custom-domain", "/custom-path"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("routeTestParameters")
+    public void testGetHostDomainPath(CloudRoute route, String expectedHost, String expectedDomain, String expectedPath) {
+        ApplicationURI applicationURI = new ApplicationURI(route);
+        assertParametersAreValid(expectedDomain, applicationURI.getDomain(), expectedDomain, route.getDomain()
+                                                                                                  .getName(),
+                                 expectedHost, applicationURI.getHost());
         Assertions.assertEquals(expectedPath, applicationURI.getPath());
     }
+
+    @Test
+    public void testGetURIParts() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"), null));
+        Map<String, Object> expectedParts = new HashMap<>();
+        expectedParts.put(SupportedParameters.HOST, "custom-host");
+        expectedParts.put(SupportedParameters.DOMAIN, "custom-domain");
+        expectedParts.put(SupportedParameters.ROUTE_PATH, null);
+        Assertions.assertTrue(Maps.difference(expectedParts, applicationURI.getURIParts())
+                                  .areEqual());
+    }
+
+    @Test
+    public void testGetURIPart() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"),
+                                                                            "/custom-path"));
+        assertParametersAreValid("custom-host", applicationURI.getURIPart(SupportedParameters.HOST), "custom-domain",
+                                 applicationURI.getURIPart(SupportedParameters.DOMAIN), "/custom-path",
+                                 applicationURI.getURIPart(SupportedParameters.ROUTE_PATH));
+        Assertions.assertNull(applicationURI.getURIPart("invalid-parameter"));
+    }
+
+    @Test
+    public void testURIPart() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"),
+                                                                            "custom-path"));
+        applicationURI.setURIPart(SupportedParameters.HOST, "custom-host-1");
+        applicationURI.setURIPart(SupportedParameters.DOMAIN, "custom-domain-1");
+        applicationURI.setURIPart(SupportedParameters.ROUTE_PATH, "/custom-path-1");
+        applicationURI.setURIPart("invalid-parameter", "value");
```

Comment:
What will this invalid parameter uri part do?

Reply:
Verifies nothing will happen if some other parameter is passed.